### PR TITLE
redirect

### PR DESCRIPTION
**Pull request by Rick-Anderson**

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@ardalis  ready

**Comment by Rick-Anderson** [comment]
@ardalis once you get this merged, you can list [@]HaoK  as an alternate reviewer. This is pretty simple security.

File: aspnetcore/security/preventing-open-redirects.md [hunk 1]
```diff
@@ -14,28 +14,32 @@ uid: security/preventing-open-redirects
...
 
 ## What is an open redirect attack?
 
-A common pattern in web applications, especially when a login is required, is to specify a redirect URL through the querystring. In this way, an unauthorized user attempting to access a particular resource can be redirected to a login page with their intended destination included in the URL string. Once they successfully authenticate, the user can be redirected to the URL they had originally requested. However, since the destination URL is simply specified in the querystring of the request, a malicious user could tamper with the querystring, allowing the site to redirect the user to an external, malicious URL. This technique is called an open redirect (or redirection) attack.
+Web applications frequently redirect users to a login page when they accesses resources that require authentication. The redirection typlically includes a returnUrl querystring parameter so that the user can be returned to the originally requested URL after they have successfully logged in. After the user authentications, they are redirected to the URL they had originally requested.
```
Comment (ardalis) [inline aspnetcore/security/preventing-open-redirects.md]: "when they access"

Comment (ardalis) [inline aspnetcore/security/preventing-open-redirects.md]: "includes a `returnUrl` querystring parameter"

Comment (ardalis) [inline aspnetcore/security/preventing-open-redirects.md]: "after the user authenticates" (not authentications)

File: aspnetcore/security/preventing-open-redirects.md [hunk 2]
```diff
@@ -14,28 +14,32 @@ uid: security/preventing-open-redirects
...
+If you navigate to the preceding link and log in, you are redirected to the site's [About page](http://nerddinner.com/Home/About). Imagine an attack designed to gain access to a user's credentials. The attacker creates a page that matched the look of the target app. The malacious site might use a similar domain name (for example, nerddiner.com, with a single 'n'). The attacker lures a user to navigate to a page requesting them to change their password, using a URL like this: ``http://nerddinner.com/Account/LogOn?returnUrl=http://nerddiner.com/Account/LogOn``. After successfully logging in, the user sees the page reload and they are prompted to log in agin. The user may think their first attempt had failed, and try again. The attacker  stores the user's credentials, and redirects them back to their original site.
```
Comment (ardalis) [inline aspnetcore/security/preventing-open-redirects.md]: "creates a page that matched" -> "creates a page that matches"

Comment (ardalis) [inline aspnetcore/security/preventing-open-redirects.md]: "malacious"

**Comment by Rick-Anderson** [comment]
@ardalis  merge and cleanup please

**Pull request merged**